core/src/raw/http_util/client.rs

```diff
@@ -78,6 +81,34 @@ impl HttpClient {
 
     /// Fetch a request in async way.
     pub async fn fetch(&self, req: Request<Buffer>) -> Result<Response<HttpBody>> {
+        self.fetcher.fetch(req).await
+    }
+}
+
+#[async_trait::async_trait]
+pub trait HttpFetch: Send + Sync {
+    /// Fetch a request in async way.
+    async fn fetch(&self, req: Request<Buffer>) -> Result<Response<HttpBody>>;
+}
+
+#[derive(Clone)]
+struct ReqwestHttpFetcher {
```

Comment:
Maybe we can implement `HttpFetch` for `reqwest::Client` directly?

Reply:
can we implement trait for other crate's struct? 😲

Reply:
We can do this in `opendal`. Users can implement via a wrapper.